src/Hedgehog/Extras/Test/New/Monad/UnitIO.hs

```diff
@@ -0,0 +1,102 @@
+{-# LANGUAGE BlockArguments #-}
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE ImportQualifiedPost #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+module Hedgehog.Extras.Test.New.Monad.UnitIO
+  ( UnitIO(..)
+  , UnitEnv(..)
+  , unitIOToTestT
+  ) where
+
+import Control.Concurrent qualified as IO
+import Control.Concurrent.STM
+import Control.Concurrent.STM qualified as STM
+import Control.Exception as IO
+import Control.Exception.Lifted qualified as CEL
+import Control.Monad.Catch (MonadCatch, MonadMask, MonadThrow)
+import Control.Monad.Error.Class (MonadError)
+import Control.Monad.Fix (MonadFix)
+import Control.Monad.Primitive (PrimMonad)
+import Control.Monad.Reader (MonadReader(..))
+import Control.Monad.Trans.Reader (ReaderT(..))
+import Control.Monad.Trans.Resource (ResourceT, MonadResource(..), runResourceT)
+import Data.Either
+import Data.Generics.Product.Any
+import Data.Maybe
+import Data.Monoid
+import HaskellWorks.Control.Monad
+import HaskellWorks.Prelude
+import Hedgehog
+import Hedgehog.Extras.Test.MonadAssertion
+import Hedgehog.Internal.Property qualified as H
+import Lens.Micro
+import Test.Tasty.Discover
+import Test.Tasty.Hedgehog (testProperty)
+
+newtype UnitEnv = UnitEnv { unUnitEnv :: TMVar (TestT IO ()) }
+
+newtype UnitIO a = UnitIO { runTestIO :: ExceptT H.Failure (ResourceT (ReaderT UnitEnv IO)) a }
```

Comment:
There's a lot of duplicated code in this module. Why not implement 
```suggestion
newtype Property2 env a = Property2 { runProperty2 :: ExceptT H.Failure (ResourceT (ReaderT env IO)) a }
```

And implement those functions just once?